### PR TITLE
Omuse refactor

### PR DESCRIPTION
**Pull request by merijn**
This PR mostly moves source files around, slightly modifies headers and changes CMakeLists dependencies to make the internal dependency graph of i-emic simpler for combining with omuse.

Also enables --oversubscribe in mpirun for the tests as my own laptop and the travis machine have <8 cores and else the tests fail because it refuses to spawn more tasks than cores.

I'm looking into how we can simplify the circular dependency between atmosphere/ocean/seaice (see #11), but haven't gotten around to it yet, this is more prep work for that.

### CODE REVIEW

File: src/seaice/SeaIce.C [hunk 1]
```diff
@@ -1,4 +1,5 @@
 #include "my_f2c.H"
+#include "my_f2c.H"
```
Comment (Sbte) [inline src/seaice/SeaIce.C]: Why is this included twice?

Comment (Sbte) [reply src/seaice/SeaIce.C]: Never mind this is removed in a later commit

Comment (merijn) [reply src/seaice/SeaIce.C]: Yeah, that snuck in during rebasing at some point, was easier to remove in the last patch, rather than edit the history again

**Comment by erik808** [comment]
Trouble on Cartesius: intel compiler does not like oversubscribe
```
test 19
      Start 19: partest_matrix_1

19: Test command: /hpc/eb/RedHatEnterpriseServer7/impi/5.1.3.181-iccifort-2016.3.210-GCC-5.4.0-2.26/bin64/mpirun "-np" "1" "--oversubscribe" "/home/emulder/Projects/i-emic/build/src/tests/test_matrix"
19: Test timeout computed to be: 9.99988e+06
19: [mpiexec@int1.bullx] match_arg (../../utils/args/args.c:254): unrecognized argument oversubscribe
19: [mpiexec@int1.bullx] HYDU_parse_array (../../utils/args/args.c:269): argument matching returned error
19: [mpiexec@int1.bullx] parse_args (../../ui/mpich/utils.c:4498): error parsing input array
19: [mpiexec@int1.bullx] HYD_uii_mpx_get_parameters (../../ui/mpich/utils.c:4819): unable to parse user arguments

```

**Comment by merijn** [comment]
I changed it to only set it on macOS

**Comment by erik808** [comment]
Cartesius is ok now. On my own machine I'm having linker issues. 
```
../coupledmodel/libcoupledmodel.so: undefined reference to `ML_Epetra::MultiLevelPreconditioner::ComputePreconditioner(bool)'
../coupledmodel/libcoupledmodel.so: undefined reference to `ML_Epetra::MultiLevelPreconditioner::MultiLevelPreconditioner(Epetra_RowMatrix const&, Teuchos::ParameterList const&, bool)'
```

To fix it I need to add trios to coupledmodel  link libs: `target_link_libraries(coupledmodel PUBLIC atmosphere ocean trios seaice)`. I don't understand why this only shows up on my installation.  

My installation is messed up anyway so I have to fix that and return to this next week.
